### PR TITLE
fix(assets): Bump minimum Go version to 1.23

### PR DESCRIPTION
**Pull request by ctlong**
### Are you submitting this PR against the [develop branch](https://github.com/cloudfoundry/cf-acceptance-tests/tree/develop)?

Yes

### What is this change about?

Bumps the minimum Go versions of all the Go assets to `1.23`, omitting the patch version due to handling issues with the Go buildpack in certain edge cases.

### Please provide contextual information.

- https://github.com/cloudfoundry/cf-acceptance-tests/commit/043a63ff66b5a67c36a14876edf9095d74b62783#commitcomment-145272348
- https://github.com/cloudfoundry/cf-acceptance-tests/pull/1195

### What version of cf-deployment have you run this cf-acceptance-test change against?

v48.4.0

### Please check all that apply for this PR:

- [ ] introduces a new test --- Are you sure everyone should be running this test?
- [ ] changes an existing test
- [ ] requires an update to a CATs integration-config

### Did you update the README as appropriate for this change?

- [ ] YES
- [x] N/A

### If you are introducing a new acceptance test, what is your rationale for including it CATs rather than your own acceptance test suite?

N/A

### How many more (or fewer) seconds of runtime will this change introduce to CATs?

None

### What is the level of urgency for publishing this change?

- [ ] **Urgent** - unblocks current or future work
- [x] **Slightly Less than Urgent**

### Tag your pair, your PM, and/or team!

None

### CODE REVIEW

File: assets/binary/go.mod [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module github.com/cloudfoundry/cf-acceptance-tests/assets/binary
 
-go 1.22
+go 1.23
 
-toolchain go1.22.0
+toolchain go1.23.7
```
[the remark this answers is not in the document]
Comment (ctlong) [reply assets/binary/go.mod:5]: The versions don't need to be the same, in fact they're frequently different. The newer [Go Toolchain](https://go.dev/doc/toolchain) logic has the `go` directive as the "the minimum required Go version necessary to use a module or workspace" and the `toolchain` directive as the "preferred toolchain that takes precedence over the go line when the go command is deciding which toolchain to use".

In our case, we're saying we _require_ at least go1.23.0 to use the module, but we _recommend_ go1.23.7 (the latest patch on that minor).

**Review by dimivel — APPROVED** [review]

File: assets/binary/go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (oliver-heinrich) [reply assets/binary/go.mod:5]: Hi @ctlong , thank you for the additional background info. CU Oliver
